st2client/st2client/client.py

```diff
@@ -129,115 +130,194 @@ def __init__(
 
         self.api_key = api_key
 
+        if basic_auth:
+            if len(basic_auth.split(":")) != 2:
+                raise ValueError(
+                    "basic_auth config options needs to be in the "
+                    "username:password notation"
+                )
+
+            self.basic_auth = tuple(basic_auth.split(":"))
```

Comment:
Why not use tuple unpacking?

```suggestion
            username, *password = basic_auth.split(":")
            password = ":".join(password)  # Put the password back together
            self.basic_auth = (username, password)
```

Reply:
That would also work, but per my comment above (https://github.com/StackStorm/st2/pull/5152#discussion_r618716121), splitting with ``maxsplit=1`` is even simple - 1 line vs 3 lines :)

Reply:
e67d40e9d